bindings/sundials4py/arkode/arkode.cpp

```diff
@@ -0,0 +1,244 @@
+void bind_arkode_mristep(nb::module_& m);
+void bind_arkode_forcingstep(nb::module_& m);
+void bind_arkode_splittingstep(nb::module_& m);
+
+// ARKODE callback binding macros
+#define BIND_ARKODE_CALLBACK(NAME, FN_TYPE, MEMBER, WRAPPER, ...)       \
+  m.def(                                                                \
+    #NAME,                                                              \
+    [](void* ark_mem, std::function<std::remove_pointer_t<FN_TYPE>> fn) \
+    {                                                                   \
+      auto fn_table    = get_arkode_fn_table(ark_mem);                  \
+      fn_table->MEMBER = nb::cast(fn);                                  \
+      if (fn) { return NAME(ark_mem, &WRAPPER); }                       \
+      else { return NAME(ark_mem, nullptr); }                           \
+    },                                                                  \
+    __VA_ARGS__)
+
+#define BIND_ARKODE_CALLBACK2(NAME, FN_TYPE1, MEMBER1, WRAPPER1, FN_TYPE2, \
+                              MEMBER2, WRAPPER2, ...)                      \
+  m.def(                                                                   \
+    #NAME,                                                                 \
+    [](void* ark_mem, std::function<std::remove_pointer_t<FN_TYPE1>> fn1,  \
+       std::function<std::remove_pointer_t<FN_TYPE2>> fn2)                 \
+    {                                                                      \
+      auto fn_table     = get_arkode_fn_table(ark_mem);                    \
+      fn_table->MEMBER1 = nb::cast(fn1);                                   \
+      fn_table->MEMBER2 = nb::cast(fn2);                                   \
+      if (fn1) { return NAME(ark_mem, &WRAPPER1, &WRAPPER2); }             \
+      else { return NAME(ark_mem, nullptr, &WRAPPER2); }                   \
+    },                                                                     \
+    __VA_ARGS__)
+
+void bind_arkode(nb::module_& m)
+{
+#include "arkode_generated.hpp"
+
+  /////////////////////////////////////////////////////////////////////////////
+  // Interface view classes for ARKODE level objects
+  /////////////////////////////////////////////////////////////////////////////
+
+  nb::class_<ARKodeView>(m, "ARKodeView")
+    .def("get", nb::overload_cast<>(&ARKodeView::get, nb::const_),
+         nb::rv_policy::reference);
+
+  /////////////////////////////////////////////////////////////////////////////
+  // ARKODE user-supplied function setters
+  /////////////////////////////////////////////////////////////////////////////
+
+  m.def("ARKodeRootInit",
+        [](void* ark_mem, int nrtfn,
+           std::function<std::remove_pointer_t<ARKRootStdFn>> fn)
+        {
+          auto fn_table    = get_arkode_fn_table(ark_mem);
+          fn_table->rootfn = nb::cast(fn);
+          return ARKodeRootInit(ark_mem, nrtfn, &arkode_rootfn_wrapper);
+        });
```

Comment:
Should this have
```suggestion
        },
        nb::arg("arkode_mem"), nb::arg("nrtfn"), nb::arg("root_fn").none());
```
similar to `ARKodeResize` below?